File: app/src/main/java/ua/meugen/android/pdfcomposer/ui/activities/base/fragment/state/BaseState.java
package ua.meugen.android.pdfcomposer.ui.activities.base.fragment.state;

import android.os.Bundle;

public class BaseState implements MvpState {

    protected Bundle bundle;

    public void attachBundle(final Bundle bundle) {
        this.bundle = bundle == null ? Bundle.EMPTY : bundle;
    }

    public void detachBundle() {
        this.bundle = null;
    }
}

File: app/src/main/java/ua/meugen/android/pdfcomposer/ui/activities/base/fragment/state/BaseState.kt
package ua.meugen.android.pdfcomposer.ui.activities.base.fragment.state

import android.os.Bundle

open class BaseState : MvpState {

    protected var bundle: Bundle? = null

    fun attachBundle(bundle: Bundle?) {
        this.bundle = bundle ?: Bundle.EMPTY
    }

    fun detachBundle() {
        this.bundle = null
    }
}
